xarray/backends/zarr.py

```diff
@@ -328,35 +358,64 @@ def open_group(
             open_kwargs["storage_options"] = storage_options
         elif storage_options:
             raise ValueError("Storage options only compatible with zarr>=2.5.0")
+
         if chunk_store:
             open_kwargs["chunk_store"] = chunk_store
+            if consolidated is None:
+                consolidated = False
 
-        if consolidated:
+        if consolidated is None:
+            try:
+                zarr_group = zarr.open_consolidated(store, **open_kwargs)
+            except KeyError:
+                warnings.warn(
+                    "failed to open Zarr store with consolidated metadata, "
+                    "falling back to try reading non-consolidated metadata. In "
+                    "the future, this will be an error instead. To skip "
+                    "this fallback, set consolidated=True; to skip attempting "
+                    "to read consolidated metadata, consolidated=False.",
+                    FutureWarning,
```

Comment:
This warning is the part that I'm least confident about here:

- It suggests that Xarray may not read non-consolidated metadata in the future ("In the future, this will be an error instead"). Given the non-official status of consolidated metadata, this may be premature? In which case, we should probably substitute `FutureWarning` -> `RuntimeWarning`
- I'm not sure the warning would be appreciated by all users. It's a nag to add an explicit argument that they may not have control over. Maybe we should also mention the likely performance cost, and suggest consolidating metadata on the existing store with `zarr.consolidate_metadata`?

Reply:
> * Given the non-official status of consolidated metadata, this may be premature?

I worry that having a library in the ecosystem that can't read a valid Zarr because of a non-specified option will amount to a fork.

Reply:
Well, arguably we've already forked Zarr due to the required `_ARRAY_DIMENSIONS` attribute :)

Reply:
That's in an even weirder spot given that netCDF-C (and ncZarr) are not currently using consolidated metadata.

Reply:
OK, I revised the warning (and docs) to take both of these into account.

Reply:
> Well, arguably we've already forked Zarr .... :)

_whistles_